src/virtualenv/activation/python/activate_this.py

```diff
@@ -5,51 +5,28 @@
 
 This can be used when you must use an existing Python interpreter, not the virtualenv bin/python.
 """
-import json
 import os
 import site
 import sys
 
 try:
-    __file__
+    abs_file = os.path.abspath(__file__)
 except NameError:
     raise AssertionError("You must use exec(open(this_file).read(), {'__file__': this_file}))")
 
-
-def set_env(key, value, encoding):
-    if sys.version_info[0] == 2:
-        value = value.encode(encoding)
-    os.environ[key] = value
-
+bin_dir = os.path.dirname(abs_file)
+base = bin_dir[: -len("__BIN_NAME__") - 1]  # strip away the bin part from the __file__, plus the path separator
 
 # prepend bin to PATH (this file is inside the bin directory)
-bin_dir = os.path.dirname(os.path.abspath(__file__))
-set_env("PATH", os.pathsep.join([bin_dir] + os.environ.get("PATH", "").split(os.pathsep)), sys.getfilesystemencoding())
-
-base = os.path.dirname(bin_dir)
-
-# virtual env is right above bin directory
-set_env("VIRTUAL_ENV", base, sys.getfilesystemencoding())
-
-# add the virtual environments site-packages to the host python import mechanism
-prev = set(sys.path)
-
-site_packages = r"""
-__SITE_PACKAGES__
-"""
-
-for site_package in json.loads(site_packages):
-    if sys.version_info[0] == 2:
-        site_package = site_package.encode("utf-8").decode(sys.getfilesystemencoding())
-    path = os.path.realpath(os.path.join(os.path.dirname(__file__), site_package))
-    if sys.version_info[0] == 2:
-        path = path.encode(sys.getfilesystemencoding())
-    site.addsitedir(path)
+os.environ["PATH"] = os.pathsep.join([bin_dir] + os.environ.get("PATH", "").split(os.pathsep))
+os.environ["VIRTUAL_ENV"] = base  # virtual env is right above bin directory
 
+# add the virtual environments libraries to the host python import mechanism
+prev_length = len(sys.path)
+for lib in "__LIB_FOLDERS__".split(os.pathsep):
```

Comment:
This is not correct. It will always cycle through `['__LIB_FOLDERS__', ]`.

Reply:
Why?

Reply:
Because you are splitting `"__LIB_FOLDERS__"` which is a string not a variable. 
You probably wanted to use something like `os.environ["__LIB_FOLDERS__"]` instead?

Reply:
You're incorrect on this; the file is a template not the actual file - https://github.com/pypa/virtualenv/blob/master/src/virtualenv/activation/python/__init__.py#L17-L25

Reply:
Huh? OK, now I feel stupid.
Sorry for bothering you. :bow: